docs/source/posts/2023/2023-01-31-final-report-mohamed.rst

```diff
@@ -0,0 +1,266 @@
+  * Allow the creation and removal of actors from the scene according to the keyframes.
+  * Visualize the motion path of positional animation.
+  * Speed the animation using GLSL vertex and fragment shaders.
+
+Modified Objectives
+-------------------
+
+
+* Adding a playback panel for controlling the timeline.
+* Billboard actor using the geometry shader.
+* Hierarchical animation support.
+* Animating primitives of the same Fury actor separately.
+* Color interpolators.
+
+Objectives Completed
+--------------------
+
+
+* 
+  Keyframes Animation API
+
+
+  * ``Animation`` Class
+  * 
+    The ``Animation`` class is the main part of the FURY animation module. It is responsible for keyframe animations for a single or a group of FURY actors.  The ``Animation`` is able to handle multiple attributes and properties of Fury actors such as position, color, scale, rotation, and opacity.  It is also capable of doing the following:
+
+
+    * Set animation keyframes and events.
+    * Animate custom properties.
+    * Support add-to-scene/remove-from-scene events.
+    * Can animate other animations (Hierarchical animation)
+    * Set or change the keyframes interpolation method.
+    * Visualize the motion path of the positional animation.
+
+
+    * ``Timeline`` Class
+      The ``Timeline`` is the player of FURY ``Animations``\ ; it controls the playback of one or more FURY animations. It also has the option to include a very useful playback panel to help control the playback of the animation. The ``Timeline`` can have a fixed length or get its duration from the animations added to it dynamically. It can loop, play once, change speed, play, pause, and stop.
+
+  * 
+    Keyframes Interpolators
+       Interpolation is also a core part of the keyframes animation. It is responsible for filling in the blanks between the keyframes so that we have transitional states between the set keyframes. Another factor is that interpolators must be super-optimized to interpolate data in a minimum amount of time as possible or else it would be the bottleneck that lags the animation. 
+       The following interpolators have been implemented as a part of the keyframes animation API:
+
+
+    * Step Interpolator
+    * Linear Interpolator
+    * Spherical Linear Interpolator (Slerp)
+    * Spline Interpolator 
+    * Cubic Spline Interpolator 
+    * Cubic Bézier Interpolator
+    * Color Interpolators:
+
+      * XYZ Color Interpolator
+      * Lab Color Interpolator
+      * HSV Color Interpolator
+
+    .. raw:: html
+
+        <iframe id="player" type="text/html"   width="440" height="390" src="https://user-images.githubusercontent.com/63170874/194437472-e9ad04b5-0d80-40bf-bd06-64952aeaa65f.mp4" frameborder="0"></iframe>
+
+    .. raw:: html
+
+        <iframe id="player" type="text/html"   width="440" height="390" src="https://user-images.githubusercontent.com/63170874/194441472-a33f66cb-4860-4659-83f7-720a69696c12.mp4" frameborder="0"></iframe>
+
```

Comment:
Is it possible to have a gif. Otherwise, videos to resize, it look strange:

![image](https://user-images.githubusercontent.com/23106443/216666117-d11b3971-85c7-421a-9cbd-45967ca6172d.png)

Reply:
@skoudoro thanks for generating the docs. I will change those to GIFs.